### ISSUE TITLE
Failed with error: Out of Range (8) read datastream of length 64 over by 1 error unpacking uint8_t

### ISSUE THREAD

**Issue by lingyiliu016** [EOSIO/eos#4268]
I have developed a smart contract (market)that has a table named auction, then I deployed it on test blockchain, then I push a action that wrote some data into the table, after I push the action success ,I check the data in auction ,some error occured! error as detail:
Here is my command line( push action ): `cleos push action xxx.acnt createauc '["10","child.acnt","10","5.0000 EOS","1529569038","我要拍卖一只精灵"]' -p child.acnt`;
Here is my command line( query data ): `cleos get table child.acnt child.acnt auction`
<img width="817" alt="20180621180223" src="https://user-images.githubusercontent.com/27277897/41712511-46ae01b4-757d-11e8-9f2d-69aa9291fe89.png">
Here is the information on nodeos
<img width="981" alt="20180621180727" src="https://user-images.githubusercontent.com/27277897/41712790-0491cc06-757e-11e8-9a33-cd1fbbdde8fa.png">

I have no idea about this issue and I don't konw how to do it?
help me ,please! Thank you very much!

**Comment by lingyiliu016** [comment]
when I clear table auction , then I carried out command line `cleos push action cly.acnt createauc '["1","cly.acnt","1","5.0000 EOS","1529569038",""]' -p cly.acnt`(chinese ommitted), and I get table information, same error occur!

**Comment by lingyiliu016** [comment]
Here is part of auction code.(the define of table auction)
<img width="961" alt="20180621194152" src="https://user-images.githubusercontent.com/27277897/41717128-41c4b2c0-758b-11e8-8d76-a2bb4b6746da.png">
Here is the action of insert data into table auction.
<img width="857" alt="20180621194408" src="https://user-images.githubusercontent.com/27277897/41717213-8b1f48b8-758b-11e8-9d42-656fad017cd7.png">

**Comment by lingyiliu016** [comment]
Here is part of my abi file that defined the Auction struct.
<img width="850" alt="20180622092923" src="https://user-images.githubusercontent.com/27277897/41753285-232423d6-75ff-11e8-9284-fde38a2d99d7.png">
<img width="483" alt="20180622092952" src="https://user-images.githubusercontent.com/27277897/41753291-2a57db8e-75ff-11e8-8f73-ba64fd41f5ad.png">
 According to the above picture, we can easily find that the defination of auction struct is the same as the define in hpp file.

**Comment by lingyiliu016** [comment]
There is an interesting phenomenon. When I clear the auction table, then I query the content of auction table, everything is normal . what's the matter ?
<img width="1023" alt="20180622094436" src="https://user-images.githubusercontent.com/27277897/41753630-257e79ea-7601-11e8-8a33-fe4fb1231368.png">

**Comment by heifner** [comment]
Two things. Your EOSLIB_SERIALIZE does not match your table. Also you are using unicode with a string which I doubt will work.

**Comment by lingyiliu016** [comment]
Thank you very much! I'm very glad that you have found the source of the problem. I found this problem that EOSLIB_SERIALIZE lack two member value : flag and memo in the defination of struct Auction almost simultaneously with you.
Now I solve this problem. Here is the result.
<img width="785" alt="20180622103547" src="https://user-images.githubusercontent.com/27277897/41755075-0d4e3dd6-7608-11e8-8d4e-ff74fc567ecb.png">
Also, I want to answer you doubts.
string type in the defination of struct is not a problem .
Finally ,Thank you very much! Although I have solved this issue on my own.
